src/test/scala/org/constellation/wallet/ValidateWalletFuncTest.scala

```diff
@@ -39,6 +40,92 @@ class ValidateWalletFuncTest extends FlatSpec {
 
   }*/
 
+  "getEncoding" should "correctly encode transaction data" in {
+    val src = "DAG48nmxgpKhZzEzyc86y9oHotxxG57G8sBBwj34"
+    val dst = "DAG48nmxgpKhZzEzyc86y9oHotxxG57G8sBBwj56"
+    val amount = 100000000L
+    val prevHash = "08e6f0c3d65ed0b393604ffe282374bf501956ba447bc1c5ac49bcd2e8cc44fd"
+    val ordinal = 567L
+    val fee = 123456L
+    val salt = 7370566588033602435L
+    val lastTxRef = LastTransactionRef(prevHash, ordinal)
+    val txData = TransactionEdgeData(amount = amount, lastTxRef = lastTxRef, fee = Option(fee), salt = salt)
+    val oe = ObservationEdge(
+      Seq(
+        TypedEdgeHash(src, EdgeHashType.AddressHash),
+        TypedEdgeHash(dst, EdgeHashType.AddressHash)
+      ),
+      TypedEdgeHash(txData.getEncoding, EdgeHashType.TransactionDataHash)
+    )
+    val soe = SignHelp.signedObservationEdge(oe)(Fixtures.tempKey)
+    val tx = Transaction(edge = Edge(oe, soe, txData), lastTxRef = lastTxRef, isDummy = false, isTest = false)
+    val expectedEncoded =
+      "02" + //number of parents
```

Comment:
If we're re-using these below consider making each part of expectedEncoded vals

Reply:
I'm reusing some of it, but I think actually that extracting common parts in this particular example can make things less clear and I would need to change from flatSpec to freeSpec hence refactor other tests to extract these - I think it's not worth it ;).